hopsanhdf5exporter/hopsanhdf5exporter.cpp

```diff
@@ -0,0 +1,187 @@
+#include <ctime>
+
+//! @brief Help function to append string attribute to HDF5 object
+void appendH5Attribute(H5::H5Object &rObject, const H5std_string &attrName, const H5std_string &attrValue)
+{
+    H5::DataSpace attr_dataspace = H5::DataSpace( H5S_SCALAR );
+    H5::StrType attr_strtype( H5::PredType::C_S1, std::max(attrValue.size(), size_t(1)) );
+    H5::Attribute attribute = rObject.createAttribute( attrName, attr_strtype, attr_dataspace );
+    attribute.write( attr_strtype, attrValue );
+}
+
+HopsanHDF5Exporter::HopsanHDF5Exporter(const hopsan::HString &rFilePath, const hopsan::HString &rModelFileName, const hopsan::HString &rToolName) :
+    mFilePath(rFilePath),
+    mModelFileName(rModelFileName),
+    mToolName(rToolName) {}
+
+void HopsanHDF5Exporter::addVariable(hopsan::HString &rSystemHierarchy, const hopsan::HString &rComponentName, const hopsan::HString &rPortName, const hopsan::HString &rVariableName, const hopsan::HString &rAliasName, const hopsan::HString &rUnit, const hopsan::HString &rQuantity, hopsan::HVector<double> &rDataVector)
+{
+    mSystemHierarchies.append(rSystemHierarchy);
+    mComponentNames.append(rComponentName);
+    mPortNames.append(rPortName);
+    mVariableNames.append(rVariableName);
+    mAliasNames.append(rAliasName);
+    mUnits.append(rUnit);
+    mQuantities.append(rQuantity);
+    mDataVectors.append(rDataVector);
+}
+
+bool HopsanHDF5Exporter::writeToFile()
+{
+    HString error;
+    const H5std_string filePath = mFilePath.c_str();
+    try {
+        // turn off auto printing of thrown exceptions so that they can be handled below
+        H5::Exception::dontPrint();
+
+        // Create and open a file
+        H5::H5File file(filePath, H5F_ACC_TRUNC);
+
+        //Generate date and time string
+        time_t rawtime;
+        struct tm * timeinfo;
+        char timestr[100];
+        time (&rawtime);
+        timeinfo = localtime(&rawtime);
+        std::strftime(timestr,sizeof(timestr),"%a %b %d %H:%M:%S %Y",timeinfo);
+        HString dateTime = HString(timestr);
+
+        H5::Group root = file.openGroup("/");
+        appendH5Attribute(root, "date", dateTime.c_str());
+        appendH5Attribute(root, "model", mModelFileName.c_str());
+        appendH5Attribute(root, "tool", mToolName.c_str());
+
+        // Build directory/group hierarchy
+        // We need this to avoid massive exception casting when creating directories as
+        // group names will be repeated, also we need to create one group depth at a time
+        // The map will sort the group names as unique values in the correct order
+        std::map<HString, char> varhier;
+        for(size_t i=0; i<mSystemHierarchies.size(); ++i) {
+
+            //Split system hierarchy string to a vector of sub strings
+            std::vector<HString> sysnames;
+            while(!mSystemHierarchies[i].empty()) {
+                sysnames.push_back(mSystemHierarchies[i].substr(0, mSystemHierarchies[i].find("."))); // token is "scott"
```

Comment:
"scott" ? dott ?
Also same thing further down

Reply:
Ops, that came from stackoverflow :)